common/activecluster/manager.go

```diff
@@ -376,5 +376,11 @@ func (m *managerImpl) getClusterSelectionPolicy(ctx context.Context, domainID, w
 		return nil, err
 	}
 
+	if plcy == nil {
```

Comment:
I think it's better to move this logic to persistence layer and have persistence tests covering this case

Reply:
it's already done by execution manager which is one level above persistence layer. if a specific db plugin returns nil row execution manager ensures it's converted to entity not found. 
I also added it here just in case. e.g. the mock in the test returns nil